File: android/app/src/main/java/com/driveincar/ui/result/ResultScreen.kt
package com.driveincar.ui.result

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.driveincar.core.time.TimeFormat

@Composable
fun ResultScreen(
    courseId: String,
    timeMs: Long,
    averageKmh: Double,
    flagged: Boolean,
    personalBest: Boolean,
    onViewRanking: () -> Unit,
    onRetry: () -> Unit,
    onBackToMap: () -> Unit,
) {
    Column(
        modifier = Modifier.fillMaxSize().padding(24.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Text("기록", style = MaterialTheme.typography.titleLarge, color = MaterialTheme.colorScheme.onSurfaceVariant)
        Spacer(Modifier.height(12.dp))
        Text(
            text = TimeFormat.raceTime(timeMs),
            fontSize = 96.sp,
            fontWeight = FontWeight.Bold,
        )
        Spacer(Modifier.height(8.dp))
        Text("평균 ${"%.0f".format(averageKmh)} km/h", color = MaterialTheme.colorScheme.onSurfaceVariant)

        Spacer(Modifier.height(24.dp))
        if (personalBest) {
            Surface(
                color = MaterialTheme.colorScheme.primary,
                contentColor = MaterialTheme.colorScheme.onPrimary,
                shape = RoundedCornerShape(8.dp),
            ) {
                Text(
                    "베스트 갱신!",
                    modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp),
                    fontWeight = FontWeight.Bold,
                )
            }
        }
        if (flagged) {
            Spacer(Modifier.height(12.dp))
            Surface(
                color = MaterialTheme.colorScheme.errorContainer,
                contentColor = MaterialTheme.colorScheme.onErrorContainer,
                shape = RoundedCornerShape(8.dp),
            ) {
                Text(
                    "비정상 평균 속도로 리더보드에서 제외됨",
                    modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp),
                )
            }
        }

        Spacer(Modifier.height(48.dp))

        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            Button(
                onClick = onViewRanking,
                modifier = Modifier.weight(1f).height(52.dp),
                shape = RoundedCornerShape(12.dp),
            ) { Text("랭킹 보기") }
            OutlinedButton(
                onClick = onRetry,
                modifier = Modifier.weight(1f).height(52.dp),
                shape = RoundedCornerShape(12.dp),
            ) { Text("다시 도전") }
        }
        Spacer(Modifier.height(8.dp))
        OutlinedButton(
            onClick = onBackToMap,
            modifier = Modifier.fillMaxWidth().height(48.dp),
            shape = RoundedCornerShape(12.dp),
        ) { Text("지도로") }
    }
}

File: ios/DriveInCar/Features/Result/ResultScreen.swift
import SwiftUI

struct ResultScreen: View {
    let courseId: String
    let timeMs: Int64
    let averageKmh: Double
    let flagged: Bool
    let personalBest: Bool
    let onViewRanking: () -> Void
    let onRetry: () -> Void
    let onBackToMap: () -> Void

    var body: some View {
        VStack(spacing: 16) {
            Spacer()
            Text("기록").font(.title3).foregroundStyle(.secondary)
            Text(TimeFormat.raceTime(ms: timeMs))
                .font(.system(size: 96, weight: .bold).monospacedDigit())
            Text("평균 \(Int(averageKmh)) km/h").foregroundStyle(.secondary)

            if personalBest {
                Text("베스트 갱신!")
                    .padding(.horizontal, 12).padding(.vertical, 6)
                    .foregroundStyle(.white)
                    .background(Brand.primary)
                    .clipShape(Capsule())
            }
            if flagged {
                Text("비정상 평균 속도로 리더보드에서 제외됨")
                    .padding(.horizontal, 12).padding(.vertical, 6)
                    .background(Color.red.opacity(0.2))
                    .clipShape(Capsule())
            }

            Spacer()

            HStack(spacing: 8) {
                Button {
                    onViewRanking()
                } label: { Text("랭킹 보기").frame(maxWidth: .infinity, minHeight: 52) }
                    .buttonStyle(.borderedProminent)
                Button {
                    onRetry()
                } label: { Text("다시 도전").frame(maxWidth: .infinity, minHeight: 52) }
                    .buttonStyle(.bordered)
            }
            Button {
                onBackToMap()
            } label: { Text("지도로").frame(maxWidth: .infinity, minHeight: 48) }
                .buttonStyle(.bordered)
        }
        .padding(24)
        .navigationBarBackButtonHidden(true)
    }
}
